cmd/internal/trender.go

```diff
@@ -64,26 +89,28 @@ func init() {
 	templateRenderCmd.Flags().StringP("instance-group-name", "g", "", "the instance-group name to render")
 	templateRenderCmd.Flags().StringP("address", "a", "", "address of the instance spec")
 	templateRenderCmd.MarkFlagRequired("address")
-	templateRenderCmd.Flags().StringSliceP("az", "z", []string{}, "az's of the instance spec")
-	templateRenderCmd.MarkFlagRequired("az")
+	templateRenderCmd.Flags().StringP("az", "z", "", "az's of the instance spec")
 	templateRenderCmd.Flags().StringP("id", "d", "", "id of the instance spec")
 	templateRenderCmd.MarkFlagRequired("id")
 	templateRenderCmd.Flags().StringP("index", "x", "", "index of the instance spec")
-	templateRenderCmd.MarkFlagRequired("index")
 	templateRenderCmd.Flags().StringP("ip", "i", "", "ip of the instance spec")
-	templateRenderCmd.MarkFlagRequired("ip")
 	templateRenderCmd.Flags().StringP("name", "m", "", "name of the instance spec")
 	templateRenderCmd.MarkFlagRequired("name")
 
 	viper.AutomaticEnv()
 	viper.BindPFlag("jobs_dir", templateRenderCmd.Flags().Lookup("jobs-dir"))
-	viper.BindPFlag("instance_group_name", templateRenderCmd.Flags().Lookup("instance-group_name"))
+	viper.BindPFlag("instance_group_name", templateRenderCmd.Flags().Lookup("instance-group-name"))
 	viper.BindPFlag("address", templateRenderCmd.Flags().Lookup("address"))
 	viper.BindPFlag("az", templateRenderCmd.Flags().Lookup("az"))
 	viper.BindPFlag("id", templateRenderCmd.Flags().Lookup("id"))
 	viper.BindPFlag("index", templateRenderCmd.Flags().Lookup("index"))
 	viper.BindPFlag("name", templateRenderCmd.Flags().Lookup("name"))
 	viper.BindPFlag("ip", templateRenderCmd.Flags().Lookup("ip"))
+
+	viper.BindEnv("az", "CF_OPERATOR_AZ")
+	viper.BindEnv("ip", "POD_IP")
+	viper.BindEnv("azi", "CF_OPERATOR_AZ_INDEX")
```

Comment:
I don't think this binding is used anywhere.

Reply:
Yes. I forgot to remove this.